core/src/components/checkbox/checkbox.ionic.scss

```diff
@@ -356,66 +356,59 @@ input {
   height: inherit;
 }
 
-// disabled, unchecked checkbox
-:host(.checkbox-disabled) .native-wrapper {
-  border-color: globals.$ionic-color-neutral-800;
-
-  background-color: globals.$ionic-state-disabled;
-}
```

Comment:
From what I recall it was because this seems unnecessary, as the mixin state-disabled was already being applied to the pseudo-element, which should be enough for the disabled styling: 
https://github.com/ionic-team/ionic-framework/blob/4d81c8f55f3f6bb2144e5a413a16077b10ff1583/core/src/components/checkbox/checkbox.ionic.scss#L352

Also, I noticed this when running the tests for the first time, the snapshot returned a grey background. I believe this was now noticeable due to the changes on the tokens:

![Screenshot 2024-10-18 at 10 13 12](https://github.com/user-attachments/assets/8219b5a8-f326-49eb-b348-60d62bc80f1c)

Let me know if I'm missing some detail here or use-case :)